docs/guide/database-insert-update.md

```diff
@@ -0,0 +1,159 @@
+
+## Inserting a new piece
+
+Inserting a new piece works very similarly to updating an existing one. The `self.insert` method takes the same arguments: `self.insert(req, piece, options)`. The big difference, of course, is that there is no existing document to find, update, and resubmit.
+
+Instead, use the `self.newInstance()` method to get a fresh document object of the proper document type. That method uses the doc type's [field schema](/guide/content-schema.md) to generate the essential document properties with any default values. We can then add any initial data to that essentially blank document object.
+
+<AposCodeBlock>
+  ```javascript
+  module.exports = {
+    extend: '@apostrophecms/piece-type',
+    // ...
+    methods (self) {
+      async addNewDog(initialInfo) {
+        // Generate a req object using the contributor role
+        const req = apos.task.getReq({ role: 'contributor' });
+        // Generate a blank dog data object.
+        let newDog = self.newInstance();
+        // Add our initial information to the object.
+        newDog = {
+          ...newDog,
+          ...initialInfo
+        };
+        // Insert the dog with the asynchronous `self.insert` method
+        const insertResult = await self.insert(req, newDog);
+
+        return insertResult;
+      }
+    }
+  };
+  ```
+  <template v-slot:caption>
+    modules/dog/index.js
+  </template>
+</AposCodeBlock>
+
+## Inserting pages
+
+As mentioned above, pages have a position within the page tree. The `insert` method for pages therefore requires additional arguments to place a new page in that hierarchy properly. Also [as described above](#updating-page-documents) we normally call the insert method directly from the `@apostrophecms/page` module, aliased in server-side code as `self.apos.page`.
+
+The arguments for the `self.apos.page.insert()` method are:
+
+| Argument | Required | Description |
+| -------- | -------- | ----------- |
+| `req` | TRUE | The associated request object. Using a provided `req` object is important for maintaining user role permissions. |
+|`_targetId` | TRUE | The `_id` of an existing page to use as a target when inserting the new page. `_home` and `_archive` are optional conveniences for the home page and [archived section](/reference/api/pages.md#moving-pages-to-the-archive), respectively. |
+|`_position` | TRUE | A numeric value will represent the zero-based child index under the `_targetId` page. `before`, `after`, `firstChild`, or `lastChild` values set the position within the page tree for the new page in relation to the target page (see `_targetId`). `before` and `after` insert the new page as a sibling of the target. `firstChild` and `lastChild` insert the new page as a child of the target. |
```

Comment:
With all this covered here, you're not that far from completing coverage of the `move` method too. Just sayin'.

Reply:
This is copied from the REST API pages. I'll be documenting `move` method in the page module's reference page and thought that'd be better referenced from there.